app/assets/javascripts/application.js

```diff
@@ -5,6 +5,26 @@
 
 window.GOVUKPrototypeKit.documentReady(() => {
   // Add JavaScript here
+
+  if (document.getElementById('calendar-slot-picker')) {
```

Comment:
The whole calendar component is within a div with the class `calendar-slot-picker`. The JavaScript to make it work only runs if this class is on the page so it shouldn't conflict with anything else.

(The JavaScript just listens for click events on the available date elements and then adds/removes classes that show hide the slot list at the bottom and add the blue circle.)